rekor-v2-spec.md

```diff
@@ -0,0 +1,241 @@
+a public key.
+
+#### 4.1.4 Canonicalization
+
+The transparency service will perform type-specific canonicalization on records
+to de-duplicate equivalent entries. An equivalent entry is one that represents
+the same signing event. Sources of duplication include encoding schemes of the
+public key or signature. For example, the transparency service must de-duplicate
+an entry that encodes the public key with DER-encoding from one that uses the
+PEM-encoding. Canonicalization must not remove any necessary information to
+validate the signature. The service must store enough information in the Merkle
+tree leaf for an interested party to perform the same record validation to
+ensure entry auditability.
+
+### 4.2 HTTP and gRPC API
+
+The Rekor v2 write API is accessible over HTTP or gRPC, and the read-only tiles
+API and checkpoint are accessible over HTTP. Refer to the [protobuf
+specification](https://github.com/sigstore/rekor-tiles/tree/main/api/proto) for
+the exact schema, and the [clients
+guide](https://github.com/sigstore/rekor-tiles/blob/main/CLIENTS.md) for
+examples.
+
+### 4.3 Checkpoint Format
+
+The Merkle tree checkpoint, accessible via a GET request to the
+`/api/v2/checkpoint` endpoint, follows the [C2SP Transparency Log
+Checkpoints](https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md)
+specification. The checkpoint MAY additionally include witness signatures as
+specified in the [Transparency Log Witness
+Protocol](https://github.com/C2SP/C2SP/blob/main/tlog-witness.md).
+
+## 5. Client Workflows
+
+### 5.1 Signers
+
+A client signing an artifact uploads the entry record, formatted as either a
+[HashedRekordRequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/hashedrekord.proto#L31C9-L31C32)
+or
+[DSSERequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/dsse.proto#L32C9-L32C24) to the
+`/api/v2/log/entries` (HTTP) or `dev.sigstore.rekor.v2.Rekor.CreateEntry` (gRPC)
+service. The client receives, as a response, a
+[TransparencyLogEntry](https://github.com/sigstore/protobuf-specs/blob/4df5baadcdb582a70c2bc032e042c0a218eb3841/protos/sigstore_rekor.proto#L94),
+which contains the latest checkpoint and an inclusion proof for the entry.
+Because entries are batched before being integrated, and because checkpoints are
+published only at regular intervals, the client may have to wait for a few
+seconds before receiving a response, depending on the configuration of the
+service.
+
+### 5.2 Artifact Verifiers
+
+A client verifying an artifact MUST NOT perform an online lookup of the entry.
+The signing client must provide the entry's checkpoint, inclusion proof,
+verification key and digest in the verification material it provides for the
+verifying client. The verifying client uses the material it is provided with and
+the public key of the log to verify the entry's inclusion in the log.
```

Comment:
Can we provide precise steps around how a client must verify a proof? 1) Verify the checkpoint signature following https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md#note-text, including origin comparison, 2) verify the inclusion proof following https://datatracker.ietf.org/doc/html/rfc6962#section-2.1.1 (or if there's another link that provides a description of proof verification from C2SP, let's use that), 3) compare the leaf hash against a provided artifact, 4) optionally verify witness signatures

Reply:
:+1: